src/contrib/debugging-sssd.rst

```diff
@@ -0,0 +1,208 @@
+passed directly to the process.
+
+.. code-block:: console
+
+    $ gdb program $(pidof sssd_nss) -ex 'handle SIG34 nostop noprint'
+
+This however will not solve the problem when the process is waiting on a
+breakpoint and therefore it eventually gets terminated by the watchdog. To solve
+this, it is recommended to set a long watchdog interval using ``timeout`` option
+in ``sssd.conf.``
+
+.. code-block:: ini
+
+    [sssd]
+    config_file_version = 2
+    services = nss, pam
+    domains = ldap.test, samba.test, ipa.test
+    user = root
+
+    [nss]
+    timeout=30000
+    ...
+
+    [pam]
+    timeout=30000
+    ...
+
+    [domain/ldap.test]
+    timeout=30000
+    ...
+
+    [domain/samba.test]
+    timeout=30000
+    ...
+
+    [domain/ipa.test]
+    timeout=30000
+    ...
+
+Attaching debugger to a running process
+***************************************
+
+There is only one process for each responder that can be distinguished by name,
+therefore it is simple to attach a debugger to the running process. For example:
+
+.. code-block::
+
+    $ sudo gdb program `pgrep sssd_nss`
+
+There can be multiple backend ``sssd_be`` processes and we need to use the
+``--domain`` parameter to distinguish between them. Therefore we want to use the
+``-f/--full`` option of the ``pgrep`` command to make it match the whole command
+line and not only the process name. The following snippet shows how to attach
+the debugger to ``ldap.test`` domain:
+
+.. code-block::
+
+    $ sudo gdb program `pgrep -f "sssd_be.+ldap.test"`
+
+.. seealso::
+
+    We created set of `gdb extensions <https://github.com/SSSD/sssd-gdb>`__ for
+    SSSD that provides pretty printers to some difficult SSSD structures.
+
+Debugging forked process from its start
+***************************************
+
+Usually, you will suffice with attaching debugger to a running process. But
+sometimes, you want to debug the process from the very beginning. This applies
+especially to the different child processes that are forked from ``sssd_be`` to
+perform various stuff like Kerberos authentication.
+
+We can use `gdbserver`_ for that which provides a remote access for ``gdb``. It
+can either listen on given device or a TCP connection which we will use in our
+examples. To install it, run:
+
+.. _gdbserver: https://sourceware.org/gdb/onlinedocs/gdb/gdbserver-man.html
+
+.. code-tabs::
+
+    .. fedora-tab::
+
+        $ sudo dnf install -y gdb-gdbserver
+
+    .. rhel-tab::
+
+        $ sudo yum install -y gdb-gdbserver
+
+    .. ubuntu-tab::
+
+        $ sudo apt-get install -y gdbserver
+
+Now, we need to create a wrapper that would execute the process in the gdbserver
+that will listen on a specific port.
+
+.. code-block:: console
+
+    $ process=/usr/libexec/sssd/krb5_child
+    $ sudo mv "$process" "$process.exe"
+    $ sudo cat << 'EOF' > $process
+    #!/bin/bash
+    exec gdbserver :1234 /usr/libexec/sssd/krb5_child.exe "$@"
```

Comment:
Not that this doesn't work and you probably just forgot it, but I think replacing the full path by the `$process` variable would make it more coherent.

`exec gdbserver :1234 $process.exe "$@"`